### PR TITLE
Add hopSize option to browser and CLI audio analyzers

### LINKED ISSUE #214: Overlapping windows
In meyda, is it possible to specify overlapping windows, eg. by having a hop length smaller than the specified buffer size?

### PR DESCRIPTION
**Pull request by odub**
Fixes #214 

CLI example:
```
> meyda ../audio/simple_piano.wav spectralCentroid --bs=4096          

=========
Meyda CLI
=========


Buffer size: 4096
Hop size: 4096
Windowing function: hanning
Will extract:
spectralCentroid 

Starting extraction...
|----------------------|
Extraction finished.

21 frames analysed.


*********spectralCentroid*********

212.29480283473072
226.14276195697147
243.5170886257632
239.29269272739805
202.82786881033803
230.65842128691256
206.99184988680878
232.1087538032745
209.12762019268362
196.79483907418867
215.40387698415668
189.71890913379028
196.88050371829087
180.90973828656496
177.69390401561895
200.69076300528846
183.54836467326388
159.66518364328704
154.62522510265816
177.14018241411935
165.11374531554597

Average spectralCentroid: 200.05462359484068

> meyda ../audio/simple_piano.wav spectralCentroid --bs=4096 --hs=2048

=========
Meyda CLI
=========


Buffer size: 4096
Hop size: 2048
Windowing function: hanning
Will extract:
spectralCentroid 

Starting extraction...
|-------------------------------------------|
Extraction finished.

42 frames analysed.


*********spectralCentroid*********

212.29480283473072
197.6732948739163
226.14276195697147
235.243410775552
243.5170886257632
248.9578178804251
239.29269272739805
235.2293400385872
202.82786881033803
206.68692553031218
230.65842128691256
238.60695129861708
206.99184988680878
210.7683196713685
232.1087538032745
212.27584212115087
209.12762019268362
187.67659126265258
196.79483907418867
203.33929926313021
215.40387698415668
185.28706008656852
189.71890913379028
162.02639521130865
196.88050371829087
203.96229597842924
180.90973828656496
174.55547427923327
177.69390401561895
227.53956675871902
200.69076300528846
205.29626081405297
183.54836467326388
188.78633964113817
159.66518364328704
161.61344543046303
154.62522510265816
149.73452138125492
177.14018241411935
232.92809787477046
165.11374531554597
194.73546263939738

Average spectralCentroid: 201.52547162625476
```

### CODE REVIEW

File: src/utilities.js [hunk 1]
```diff
@@ -141,7 +141,7 @@ export function createMelFilterBank(numFilters, sampleRate, bufferSize) {
     // pre-populating the arrays with 0s.
     filterBank[j] = Array.apply(
             null,
-						new Array((bufferSize / 2) + 1)).map(Number.prototype.valueOf, 0);
+            new Array((bufferSize / 2) + 1)).map(Number.prototype.valueOf, 0);
```
Comment (jakubfiala) [inline src/utilities.js]: omg what were we thinking

File: src/utilities.js [hunk 2]
```diff
@@ -157,3 +157,20 @@ export function createMelFilterBank(numFilters, sampleRate, bufferSize) {
...
+    throw new Error('Buffer is too short for frame length');
+  }
+  if (hopLength < 1) {
+    throw new Error('Hop length cannot be less that 1');
+  }
+  if (frameLength < 1) {
+    throw new Error('Frame length cannot be less that 1');
+  }
+
+  var numFrames = 1 + Math.floor((buffer.length - frameLength) / hopLength);
```
Comment (jakubfiala) [inline src/utilities.js]: this could be a `const`

**Comment by jakubfiala** [comment]
let's make sure the build doesn't fail though. Also, could we have some tests for whether the frame calculation works correctly with hop size?

**Comment by odub** [comment]
@jakubfiala The build troubles seems like they'll need a codeclimate admin to work through: 
```No CODECLIMATE_REPO_TOKEN found. A CODECLIMATE_REPO_TOKEN must be specified as an environment variable.```

File: src/utilities.js (hunk 2 above)
Comment (2xAA) [reply src/utilities.js]: Considering everything else uses `var` or `let` probably keep it as `var` for now.
We should probably add some sort of linting and bump everything to `ES2015+` soon however.

Comment (2xAA) [reply src/utilities.js]: Ah, didn't see this was already amended, not a problem - I'll create an issue to lint and update the code

**Comment by odub** [comment]
@jakubfiala Ready to merge, I think

**Comment by jakubfiala** [comment]
awesome, this has been approved, going to merge now 🎉 @odub really sorry for the long wait!

**Pull request merged**